src/main/java/essenmakanan/recipe/RecipeList.java

```diff
@@ -9,32 +9,32 @@ public RecipeList() {
         recipes = new ArrayList<>();
     }
 
-    public void addRecipe(String input) {
-        String recipeTitle = input.replace("r/", "");
-        recipes.add(new Recipe(recipeTitle));
+    public void addRecipe(Recipe recipe) {
+        recipes.add(recipe);
+        assert getRecipe(recipes.size() - 1).getTitle().equals(recipe.getTitle())
+                : "Recipe is not successfully added into the list.";
     }
 
     public void addRecipe(String title, String[] steps) {
         recipes.add(new Recipe(title, steps));
     }
 
-    public ArrayList<Recipe> getRecipes() {
-        return recipes;
-    }
-
     public void deleteRecipe(Recipe recipe) {
         recipes.remove(recipe);
     }
 
     public void viewAllRecipes() {
         int count = 1;
         for (Recipe recipe : recipes) {
+            assert getRecipe(count - 1).getTitle().equals(recipe.getTitle())
+                    : "Title is not matching with the current index";
+
             System.out.println(count + ". " + recipe);
-            count ++;
+            count++;
         }
     }
 
-    public Recipe getRecipeByIndex(int index) {
+    public Recipe getRecipe(int index) {
         return recipes.get(index);
     }
```

Comment:
Should use return recipes.getRecipe(index) instead because it is one of the functions in RecipeList.